tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/tracee/pass/BridgingPassImpl.scala

```diff
@@ -0,0 +1,24 @@
+package at.forsyte.apalache.tla.tracee.pass
+
+import at.forsyte.apalache.infra.passes.Pass.PassResult
+import at.forsyte.apalache.tla.lir._
+import com.google.inject.Inject
+import com.typesafe.scalalogging.LazyLogging
+
+/**
+ * The implementation of a bounded model checker with SMT.
+ *
+ * @author
+ *   Jure Kukovec
+ */
+class BridgingPassImpl @Inject() extends BridgingPass with LazyLogging {
+
+  override def name: String = "BridgingPass"
+
+  override def execute(module: TlaModule): PassResult = Right(module)
+
+  override def dependencies = Set()
+
+  // Required by BMC, set by a pass that isn't needed in TraceE mode.
+  override def transformations = Set(ModuleProperty.Analyzed)
```

Comment:
What pass is setting this and what is it needed for?
Why can't we refactor BMC instead of hacking this in?

Reply:
This is set by `AnalysisPass`, indicating things like e.g. skolemization being properly applied IIRC.
In a normal run of BMC, with generic TLA input, we want (need) the analysis pass to run. In the tracee case, the input module is modified in such a way that  there's no expansion or skolemization anyway.
If we refactor BMC, we lose this safeguard with the normal `check`. The alternative is plugging `AnalysisPass` back in, even though it does nothing for the tracee case.

Reply:
So BMC has a safeguard checking what ran before?
I'm fine with keeping this as-is, if we can improve the comment to point out where that happens, and that we fake a superfluous `AnalysisPass` for Tracee.